addon/webextension/background/main.js

```diff
@@ -30,10 +30,14 @@ window.main = (function () {
 
   chrome.browserAction.onClicked.addListener(function(tab) {
     if(tab.url.match(/about:(newtab|blank)/i)) {
-      sendEvent("goto-myshots", "about-newtab");
+      catcher.watchPromise(analytics.refreshTelemetryPref().then(() => {
```

Comment:
Hmm. I thought preference observers were needed to minimize the performance impact of repeatedly checking prefs, but I don't see that in the docs. The only [documented](https://developer.mozilla.org/en-US/Add-ons/Code_snippets/Preferences#Using_preference_observers) reason to use a preference observer is to detect changes in prefs that normally require a restart to be toggled.

Hi @Mossop, could you weigh in on the performance cost associated with Page Shot using `browser.runtime` messaging to check a pref each time the Page Shot toolbar button is pressed? I thought that repeatedly accessing prefs caused a significant performance hit, but that information might have been wrong or out of date, as I don't see it documented anywhere on MDN.